fe4u-boilerplate-main/src/js/Popup/popupTeacherCard.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable */
+const L = require('leaflet');
+const {content, popupModalInfo} = require('../Lab2/PopupFunctional')
+
+class popupTeacherCard {
+    constructor() {
+        this.teacher = null;
+        this.favCallBack = null;
+        this.isCliked = false;
+        this.favbtn = document.getElementById('ifavorite');
+        this.closebtn = document.getElementById('close-popup')
+        this.mapbtn = document.getElementById('map-btn')
+
+        this.infoteacherPopup = document.getElementById('infoPopup');
+        this.headerPopup = document.getElementById('headerPopup');
+        this.mapPopup = document.getElementById('map-content');
+        this.favbtn.onclick = () => {
+            if (!this.teacher) return;
+            this.teacher.favorite = !this.teacher.favorite;
+            this.teacher.favorite ? this.favbtn.classList.add('selected') : this.favbtn.classList.remove('selected');
```

Comment:
You may use ternary operator only when set some value:
```
const someValue = condition ? value1 : value2;
```
As there's no setting here, you should use `if`:
```
if (this.teacher.favorite) {
  this.favbtn.classList.add('selected');
} else {
  this.favbtn.classList.add('selected')
}
```

Or you may set some variable:
```
const action = this.teacher.favorite ? 'add' : 'remove';
this.favbtn.classList[action]('selected');
```
Or just `this.favbtn.classList.toggle('selected')`